core/src/collation.rs

```diff
@@ -65,6 +65,7 @@ impl Collatable for ast::Literal {
                 bits.to_be_bytes().to_vec()
             }
             ast::Literal::Boolean(b) => vec![*b as u8],
+            ast::Literal::Placeholder => panic!("Placeholders should be replaced before collation"),
```

Comment:
I am not in love with the panic(s) here.  Probably fine for now I suppose.

Reply:
Replaced almost all of the panic/unimplemented invocations, except for one in reactor which I will remove later.